otj-metrics-core/src/main/java/com/opentable/metrics/common/SortedEntry.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.opentable.metrics.common;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Predicate;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.ComparisonChain;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+
+import com.opentable.metrics.http.HealthController;
+import com.opentable.metrics.ready.ReadyController;
+import com.opentable.metrics.ready.Result;
+
+/**
+ * Common code for both Ready and Health checks.
+ * @param <T>
+ */
+public class SortedEntry<T> implements Comparable<SortedEntry<T>> {
+    private static final  Predicate<Result> READY_CHECK_PREDICATE = Result::isReady;
+    private static final Predicate<HealthCheck.Result> HEALTH_CHECK_PREDICATE = HealthCheck.Result::isHealthy;
+    private static final Comparator<Result> READY_CHECK_COMPARATOR = ReadyController::compare;
+    private static final Comparator<HealthCheck.Result> HEALTH_CHECK_COMPARATOR = HealthController::compare;
+
+    private final String name;
+    private final T result;
+    private final Comparator<T> comparator;
+
+
+    public static Map<SortedEntry<Result>, Result> ready(boolean all, Map<String, Result> raw) {
```

Comment:
This class could also use JavaDoc's on the methods

Reply:
Yes, and when you wrote this code, you probbably should have done so ;)